src/memoize.js

```diff
@@ -0,0 +1,20 @@
+export function memoize1(compute) {
+  let cacheValue;
+  let cacheKey = [];
+  return (...keys) => {
+    let cached = cacheKey.length === keys.length;
+    if (cached) {
+      for (let i = 0; i < cacheKey.length; ++i) {
+        if (cacheKey[i] !== keys[i]) {
+          cached = false;
+          break;
+        }
+      }
+    }
+    if (!cached) {
```

Comment:
```suggestion
    if (cacheKey.length !== keys.length || cacheKey.some((k, i) => k !== keys[i])) {
```

is there any drawback to using array.some or array.every?

Reply:
Simplified as you suggested. 👍